torch/csrc/jit/codegen/cuda/lower_index_hoist.h

```diff
@@ -2,147 +2,90 @@
-      const std::unordered_map<IterDomain*, Val*>& loop_index_map,
-      const std::vector<kir::ForLoop*>& loops);
-
-  const IterDomain* concreteIndexedId() const {
-    return concrete_indexed_id_;
-  }
-
-  const std::vector<kir::ForLoop*>& usedLoops() const {
-    return used_loops_;
-  }
-
-  const std::vector<Val*>& loopIndexVals() const {
-    return loop_index_vals_;
-  }
-
-  bool operator==(const CommonIndexKey& other) const;
-
-  std::string toString() const;
-
- private:
-  //! Concrete domain of indexed domain
-  IterDomain* concrete_indexed_id_ = nullptr;
-  //! Loops used for the index
-  std::vector<kir::ForLoop*> used_loops_;
-  //! Loop index vals for the used loops
-  std::vector<Val*> loop_index_vals_;
-};
-
-struct CommonIndexKeyHash {
-  std::size_t operator()(const CommonIndexKey& key) const {
-    auto h = std::hash<const IterDomain*>{}(key.concrete_indexed_id_);
-    // NOTE: do not use other fields as the pointers can be different
-    // even when two keys can share the same index
-    return h;
-  }
-};
-
-//! Map to hold hoisted common indices
 class TORCH_CUDA_CU_API CommonIndexMap {
  public:
-  //! Register an indexd consumer domain to hoist
-  //!
-  //! Returns a corresponding hoisted index and a flag indicating if a
-  //! new index is inserted.
-  //!
-  //! Consumer domains are used even for producer indexing since
-  //! producer domains in producer indexing are temporary replay
-  //! domains.
-  std::pair<Val*, bool> insert(
-      IterDomain* indexed_consumer_id,
-      TensorDomain* consumer_td,
-      TensorDomain* ref_td,
-      const std::unordered_map<IterDomain*, Val*>& ref_index_map,
-      const std::vector<kir::ForLoop*>& loops,
-      Val* index);
-
-  //! Duplicate of above, but without a reference domain. TODO: Remove other
-  //! implementation.
-  std::pair<Val*, bool> insert(
-      IterDomain* indexed_consumer_id,
-      TensorDomain* consumer_td,
-      const std::vector<IterDomain*>& loop_domains,
-      const std::unordered_map<IterDomain*, Val*>& loop_index_map,
-      const std::vector<kir::ForLoop*>& loops,
-      Val* index);
-
-  const auto& commonIndexMap() const {
-    return common_index_map_;
-  }
-
-  const auto& useCounts() const {
-    return use_counts_;
-  }
+  //! For the given index, insert its subexpressions to the loop that has
+  //! minimum amount of computation. For example, if I have a loop
+  //! FOR i1
+  //!   FOR i2
+  //!     FOR i3
+  //!       FOR i4
+  //!         index = ((i1*1 + i2*2) + i3*3) + i4*4
+  //! then this function will try insert i1*1 to common_index_map_[FOR i1],
+  //! try insert i1*1 + i2*2 to common_index_map_[FOR i2],
+  //! try insert ((i1*1 + i2*2) + i3*3) to common_index_map_[FOR i3],
+  //! try insert ((i1*1 + i2*2) + i3*3) + i4*4 to common_index_map_[FOR i4],
+  //! Before insertion, this function recursively uses
+  //! eliminateCommonSubexpression to find existing expressions/subexpressions
```

Comment:
The function was renamed?

Reply:
Yes, it was renamed to `reuseIndexIfAlreadyComputed`. I have fixed the comment. Thanks for catching it!